src/proto/streams/flow_control.rs

```diff
@@ -173,12 +178,15 @@ impl FlowControl {
             self.available
         );
 
-        // Ensure that the argument is correct
-        assert!(self.window_size >= sz as usize);
+        // If send size is zero it's meaningless to update flow control window
```

Comment:
Is this change truly needed? If the value is zero, the assert and subtractions shouldn't do anything harmful, right?

Reply:
Oh true, very good catch. I had forgotten about negative sizes.